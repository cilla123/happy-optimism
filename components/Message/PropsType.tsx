export default interface PropsType {
  theme?: 'primary' | 'success' | 'warning' | 'error';
  size?: 'lg';
  icon?: any;
  hasArrow?: boolean;
  hasClosable?: boolean;
  onClick?: () => void;
}
