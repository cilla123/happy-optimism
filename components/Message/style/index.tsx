import '../../style/';
import '../../Icon/style/';
import './index.scss';
