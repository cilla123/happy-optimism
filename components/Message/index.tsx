import React, { PureComponent } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';
import Icon from '../Icon';

export interface MessageProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Message extends PureComponent<MessageProps, any> {

  static defaultProps = {
    prefixCls: 'za-message',
    theme: 'primary',
    hasArrow: false,
    hasClosable: false,
  };

  constructor(props) {
    super(props);
    this.state = {
      visible: true,
    };
  }

  onClick = () => {
    const { hasArrow, onClick } = this.props;
    if (hasArrow && typeof onClick === 'function') {
      onClick();
    }
  }

  render() {
    const { prefixCls, className, theme, size, icon, hasArrow, hasClosable, children } = this.props;

    const classes = classnames(`${prefixCls}`, className, {
      [`theme-${theme}`]: !!theme,
      [`size-${size}`]: !!size,
    });

    const iconRender = icon && <div className={`${prefixCls}-icon`}>{icon}</div>;
    const renderCloseIcon = hasClosable && <Icon type="wrong" onClick={() => { this.setState({ visible: false }); }} />;
    const renderArrow = hasArrow && <Icon type="arrow-right" />;
    const noFooter = !hasClosable && !hasArrow;

    return this.state.visible && (
      <div className={classes} onClick={this.onClick}>
        <div className={`${prefixCls}-header`}>{iconRender}</div>
        <div className={`${prefixCls}-body`}>{children}</div>
        {!noFooter && <div className={`${prefixCls}-footer`}>{renderArrow}{renderCloseIcon}</div>}
      </div>
    );
  }
}
