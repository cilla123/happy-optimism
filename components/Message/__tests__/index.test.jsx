import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Message from '../index';

describe('Message', () => {
  it('renders correctly', () => {
    const wrapper = render(<Message>foo</Message>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('theme', () => {
    const wrapper = render(<Message theme="error">foo</Message>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('icon', () => {
    const wrapper = render(<Message icon={<img alt="" src="https://zhongantecheng.github.io/zarm/images/state.18e78939.png" />}>foo</Message>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('hasArrow', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Message hasArrow onClick={onClick}>foo</Message>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.simulate('click');
    expect(onClick).toBeCalled();
  });

  it('hasClosable', () => {
    const wrapper = shallow(<Message hasClosable>foo</Message>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('Icon').simulate('click');
    expect(wrapper.state('visible')).toEqual(false);
  });
});
