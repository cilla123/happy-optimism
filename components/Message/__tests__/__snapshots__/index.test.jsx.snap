// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message hasArrow 1`] = `
<div
  className="za-message theme-primary"
  onClick={[Function]}
>
  <div
    className="za-message-header"
  />
  <div
    className="za-message-body"
  >
    foo
  </div>
  <div
    className="za-message-footer"
  >
    <Icon
      prefixCls="za-icon"
      type="arrow-right"
    />
  </div>
</div>
`;

exports[`Message hasClosable 1`] = `
<div
  className="za-message theme-primary"
  onClick={[Function]}
>
  <div
    className="za-message-header"
  />
  <div
    className="za-message-body"
  >
    foo
  </div>
  <div
    className="za-message-footer"
  >
    <Icon
      onClick={[Function]}
      prefixCls="za-icon"
      type="wrong"
    />
  </div>
</div>
`;

exports[`Message icon 1`] = `
<div
  class="za-message theme-primary"
>
  <div
    class="za-message-header"
  >
    <div
      class="za-message-icon"
    >
      <img
        alt=""
        src="https://zhongantecheng.github.io/zarm/images/state.18e78939.png"
      />
    </div>
  </div>
  <div
    class="za-message-body"
  >
    foo
  </div>
</div>
`;

exports[`Message renders correctly 1`] = `
<div
  class="za-message theme-primary"
>
  <div
    class="za-message-header"
  />
  <div
    class="za-message-body"
  >
    foo
  </div>
</div>
`;

exports[`Message theme 1`] = `
<div
  class="za-message theme-error"
>
  <div
    class="za-message-header"
  />
  <div
    class="za-message-body"
  >
    foo
  </div>
</div>
`;
